src/VirtoCommerce.ExperienceApiModule.Core/Index/ElasticSearchRequestBuilder.cs

```diff
@@ -106,22 +121,29 @@ public IRequestBuilder AddTerms(IEnumerable<string> terms)
                             Values = item[1].Split(valuesDelimeter, StringSplitOptions.RemoveEmptyEntries)
                                 .Select(x => x?.Replace(commaEscapeString, ","))
                                 .ToArray()
-                        }).ToArray();
-                ((AndFilter)SearchRequest.Filter).ChildFilters.AddRange(termsFields);
+                        }).ToArray<IFilter>();
+
+                AddFiltersToSearchRequest(termsFields);
             }
 
             return this;
         }
 
+        private void AddFiltersToSearchRequest(IFilter[] filters)
+        {
+            ((AndFilter)SearchRequest.Filter).ChildFilters.AddRange(filters);
+        }
+
         private void ParseFilters(string filterPhrase)
         {
+            var filters = new List<IFilter>();
+
             if (filterPhrase == null)
             {
                 return;
             }
```

Comment:
Can you move filters initialization after guardian?

Reply:
Done